cpp/cmake/thirdparty/get_gputreeshap.cmake

```diff
@@ -27,6 +27,41 @@ function(find_and_configure_gputreeshap)
             GIT_TAG         ${PKG_PINNED_TAG}
     )
 
+    if (GPUTreeShap_ADDED)
+        install(TARGETS GPUTreeShap
+                EXPORT  gputreeshap-exports)
+
+        # generate gputreeshap-targets.cmake for install dir
+        rapids_export(INSTALL GPUTreeShap
+            EXPORT_SET gputreeshap-exports
+            GLOBAL_TARGETS GPUTreeShap
+            NAMESPACE GPUTreeShap::)
+
+        # generate gputreeshap-targets.cmake for binary dir
+        rapids_export(BUILD GPUTreeShap
+            EXPORT_SET gputreeshap-exports
+            GLOBAL_TARGETS GPUTreeShap
+            NAMESPACE GPUTreeShap::)
+
+        # clear out incorrect INTERFACE_SOURCES
+        set_target_properties(GPUTreeShap PROPERTIES INTERFACE_SOURCES "")
+        get_target_property(all_includes GPUTreeShap INTERFACE_INCLUDE_DIRECTORIES)
+        # clear out incorrect INTERFACE_INCLUDE_DIRECTORIES
+        set_target_properties(GPUTreeShap PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "")
+        # set INTERFACE_INCLUDE_DIRECTORIES appropriately
+        target_include_directories(GPUTreeShap INTERFACE
+            $<BUILD_INTERFACE:${all_includes}>
+            $<INSTALL_INTERFACE:include>)
```

Comment:
You will want to have `include(GNUInstallDirs)` before the `install` command and use `CMAKE_INCLUDE_DIR` as this so we can handle when a user has requested cuml to be installed into a custom include directory location